django_mongodb_backend/schema.py

```diff
@@ -351,6 +352,35 @@ def _remove_field_index(self, model, field, column_prefix=""):
             )
         collection.drop_index(index_names[0])
 
+    def _check_expression_indexes_applicable(self, expressions):
+        return all(
+            isinstance(expression.expression if isinstance(expression, OrderBy) else expression, F)
+            for expression in expressions
+        )
+
+    def _unique_supported(
+        self,
+        condition=None,
+        deferrable=None,
+        include=None,
+        expressions=None,
+        nulls_distinct=None,
+    ):
```

Comment:
I didn't analyze this complete, but it could be problematic to override this without also overriding `Index.check`. This method will silently ignore unsupported indexes. The system check framework is what gives the user a warning that the index they declared isn't supported.